extensions/amp-bind/0.1/validator-amp-bind.protoascii

```diff
@@ -46,15 +46,46 @@ tags: {  # <amp-state> (json)
   }
   spec_url: "https://www.ampproject.org/docs/reference/components/amp-bind"
 }
-tags: {  # <amp-state>
+tags: {  # <amp-state> with json child
   html_format: AMP
   tag_name: "AMP-STATE"
+  spec_name: "amp-state"
   satisfies: "amp-state"
   requires: "amp-bind extension .js script"
   disallowed_ancestor: "AMP-SIDEBAR"
   attrs: {
     name: "id"
     mandatory: true
   }
+  child_tags: {
+    mandatory_num_child_tags: 1
+    first_child_tag_name_oneof: "SCRIPT"
+  }
+  spec_url: "https://www.ampproject.org/docs/reference/components/amp-bind"
+}
+tags: {  # <amp-state> with src
```

Comment:
No, I think two TagSpec's is the best way to achieve this currently. This is a fairly specific type of constraint that I don't think we have elsewhere.

One bit problem you might have is that for an improperly formed `<amp-state>` tag, the error messages might end up referencing rules from the wrong version of the TagSpec, producing confusing/misleading errors.

One way to do this would be to add a mandatory attribute that is simply there to disambiguate meaning, ie:

`<amp-state type=embedded>`

Then, the rule would look like:

```
attrs: {
  name: 'type'
  value: 'embedded'
  mandatory: true
  dispatch_key: true
}
```

Which would guarantee that we only use this tagspec for matching when producing errors. `dispatch_key` guarantees a specific tagspec.

Not sure if that's something you want though.